src/Entity/Catalog.php

```diff
@@ -48,6 +48,8 @@ class Catalog implements CatalogInterface
 
     protected ?string $sortingType;
 
+    protected array $products = [];
```

Comment:
You don't have to change this, but maybe this could be handled by normalizer
https://symfony.com/doc/3.4/components/serializer.html#normalizers